### DISCUSSION TITLE
[Feature request/UI feedback] Adwaita: Context-sensetive window titlebar tint

### DISCUSSION THREAD

**Discussion by commie-caden** [ghostty-org/ghostty#3947] [Feature Requests, Ideas]
One of my favorite features of the LibAdwaita-based GNOME Console, Ptyxis, and the GTK4 versions of GNOME Terminal is that they provide some useful at-a-glance context clues using color. A good example is when a terminal is running as root. The whole topbar will turn red to indicate that the console is root. Also, it turns purple when using SSH or other remote software to indicate that you are operating remotely.

Small details like this go a long way to making the experience feel more intuitive. Furthermore, as every other terminal built using GTK4/LibAdwaita for the front-end has already set an expectation, it does not feel like a native app and more like a port without this feature.

![Screenshot From 2024-12-29 14-13-28](https://github.com/user-attachments/assets/cc7c396b-f33e-4467-8e62-fb6715d44bec)

*Pictured: Left is Ghostty, right is GNOME Console*

**Comment by mitchellh** [comment]
This was requested already at some point. Please find that dupe, I think @tristan957 posted this.

**Reply by tristan957** [reply]
I can't find anything.

**Reply by mitchellh** [reply]
Reopened for now... I swear there's something that asks to inspect the process tree.

**Reply by tristan957** [reply]
I might've used the wrong search terms.

**Reply by commie-caden** [reply]
If it was already posted, I do apologize. I tried finding it but couldn't before posting here.

**Comment by pluiedev** [comment]
This should be either implemented on the shell integration side, or by tools that *specifically* emit escape sequences as a part of warning the user what they're doing (great example being `run0`). IIRC there was a proposal for a standard escape code to style the terminal differently when in a "dangerous" environment but I'm not that great at Google-fu to track it down. (Or maybe it's on some freedesktop.org platform that's terrible at search engine optimization)

**Reply by mitchellh** [reply]
We can implement this using process tree inspection and I think we should. That's also a standard feature of macOS Terminal.app (via process tree inspection) and can be extended via config to support other processes too, i.e. the feature below which has been requested and something I'd like to implement as well.

![CleanShot 2025-02-17 at 13 57 31@2x](https://github.com/user-attachments/assets/63479a9d-5088-4ac1-8067-87d0c80fe367)

**Reply by miekg** [reply]
I'm opting into a `docker` group (using sg(1)), it would be nice if I could manually color the header bar as well - as there is no (long) running process that ghostty can detect

**Comment by rishubn** [comment]
+1
